Assets/Global/Scripts/Enemies/RangedEnemy/RangedEnemyStates/RangeChasingState.cs

```diff
@@ -0,0 +1,101 @@
+using UnityEngine;
+using UnityEngine.AI;
+
+namespace EnemiesNS
+{
+    public class RangeChasingState : BaseChasingState
+    {
+        private bool isMovingToRandomPosition = false;
+        public RangeChasingState(RangedEnemy enemy) : base(enemy) { }
+
+        public override void Enter()
+        {
+            enemy.animator.SetBool("Walk", true);
+            base.Enter();
+        }
+
+        public override void Exit()
+        {
+            enemy.animator.SetBool("Walk", false);
+            base.Exit();
+        }
+
+
+        private float timeSinceLastRandomMove = 0f;
+        private float randomMoveCooldown = 2f; // Cooldown time in seconds
+
+        public override void Update()
+        {
+            if (!enemy.isWaiting)
+            {
+                if (enemy.agent.isStopped)
+                    enemy.FreezeMovement(false);
+
+                if (isMovingToRandomPosition)
+                {
+                    // Check if the enemy has reached the random position
+                    if (!enemy.agent.pathPending && enemy.agent.remainingDistance <= enemy.agent.stoppingDistance)
+                    {
+                        isMovingToRandomPosition = false; // Reset the flag
+                        timeSinceLastRandomMove = Time.time; // Start cooldown
+                        Debug.Log("Reached Random Destination.");
+                        CheckState(); // Allow state change here
+                    }
+                }
+                else if (IsWithinAttackRange())
+                {
+                    // Prevent random movement if cooldown hasn't expired
+                    if (!isMovingToRandomPosition && Time.time >= timeSinceLastRandomMove + randomMoveCooldown)
+                    {
+                        Vector3 randomDestination = GetRandomNavMeshPosition(enemy.transform.position, enemy.attackRange / 2);
```

Comment:
Why is the range half of attackRange, seems random?
preferably any value that might give designers choices/levers to customize and finetune behavior should be it's own variable.

Reply:
that is not random, its supposed to be half because i don't want to run everywhere in the attack range. i want it to be half because it needs a little move not alot.